experimental/dds/tree2/src/shared-tree-core/branch.ts

```diff
@@ -315,12 +310,9 @@ export class SharedTreeBranch<TEditor extends ChangeFamilyEditor, TChange> exten
 		const change =
 			inverses.length > 0 ? this.changeFamily.rebaser.compose(inverses) : undefined;
 
-		const taggedChange = change === undefined ? undefined : makeAnonChange(change);
 		const changeEvent = {
 			type: "remove",
-			get change() {
-				return taggedChange;
-			},
+			change: change === undefined ? undefined : makeAnonChange(change),
```

Comment:
I recently learned that you can use `&&` for this.
`change && makeAnonChange(change)`

I'm not necessarily suggesting you do it here, but I'm curious if you knew about it. `??` works for "pick the thing if it's _not_ undefined, otherwise the backup thing" whereas `&&` works for "pick the thing if it _is_ undefined, otherwise the backup thing". I kind of want to bless this as a pattern that we support (specifically for handling `undefined`); it seems nice.

Reply:
I'm glad you bring this up. I first encountered that pattern in Ruby and I forget why I stopped using it. It might have something to do with JS being weird around falsy things, but I really can't remember. If it is safe to use then I'd love to use it more. Let's make it happen!

Reply:
Yea, looking at the docs I guess the way it works is that && _checks_ the truthiness/falsiness of the operand, but it _returns_ the chosen operand itself (it does _not_ convert the returned operand to true/false). So, any truthy/falsy surprises would result from unexpected coercion of the operand for the purpose of checking, but we don't have to worry about the result getting transformed unexpectedly. But yea, I guess that can be dangerous. For example:

`x && process(x)`

If the type of x is `T | undefined` (or `T | null`) - we're good... as long as T is always truthy. Which is not a given for strings, numbers, or booleans. But is true for functions, classes, objects (except the special [document.all](https://developer.mozilla.org/en-US/docs/Glossary/Falsy) object that I just learned about, lmao).

So I guess that would be the guidance. You can do this pattern for objects, and anything else you need to be very careful.